File: frontend/src/types/health.ts
export interface HealthResponse {
  status: "ok";
  service: string;
  environment: string;
  version: string;
}

export interface DependencyStatus {
  connected: boolean;
  detail: string | null;
}

export interface ReadinessResponse {
  status: "ready" | "degraded";
  postgres: DependencyStatus;
  redis: DependencyStatus;
}

File: backend/app/schemas/health.py
"""Response schemas for the health endpoints."""

from typing import Literal

from pydantic import BaseModel


class HealthResponse(BaseModel):
    status: Literal["ok"]
    service: str
    environment: str
    version: str


class DependencyStatus(BaseModel):
    connected: bool
    detail: str | None = None


class ReadinessResponse(BaseModel):
    status: Literal["ready", "degraded"]
    postgres: DependencyStatus
    redis: DependencyStatus
